### DISCUSSION TITLE
Clear files in homebrew/Caskroom

### DISCUSSION THREAD

**Discussion by xrisk** [Homebrew/discussions#4384] [Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.11-95-gd15f571
ORIGIN: https://github.com/Homebrew/brew
HEAD: d15f571eb6a994e7ca689721a54c5c6bff4219a0
Last commit: 3 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 0d4aeeb72d03e04b77ea6f6227ae0238019151ad
Core tap last commit: 12 days ago
Core tap branch: master
Core tap JSON: 06 Apr 10:37 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.zdSY3CK2hZ/org.xquartz:0
HOMEBREW_EDITOR: nvim
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.40.0 => /opt/homebrew/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 13.3-arm64
CLT: 14.3.0.0.1.1679647830
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Installers in `/opt/homebrew/Caskroom` are taking up quite a bit of space on my disk.

```
18:06:13 ❯ du -hd 1 | sort -rh
6.7G	.
5.1G	./mactex
972M	./ghidra
207M	./microsoft-teams
182M	./dotnet-sdk
 86M	./r
 62M	./powershell
 21M	./ngrok
...
```

Neither `brew cleanup -s` nor `rm $(brew --cache)` remove these. Is this expected behavior?

### Relevant casks

N/A

**Comment by SMillerDev** [comment]
Those are the apps you have installed, they are only removed by `brew rm --cask mactex` for example

**Reply by xrisk** [reply]
I understand, but the directory only contains, at most, the installer (which most people don’t need usually?) or even just a symlink to the actual app in /Applications. This seems wasteful to me.

**Comment by bjornkonig** [comment]
This did the job for me:
`find "$(brew --prefix)/Caskroom" -type f '(' -name '*.dmg' -or -name '*.pkg' ')' -delete
`
would be cool if `brew cleanup --prune=all` or similar would delete the packages in the Caskroom

**Comment by jucor** [comment]
This would indeed be very appreciated. Manually removing with this command feels a bit rustic 😅

**Reply by osalbahr** [reply]
Does the `--zap` flag do what you expect? For example, `brew rm --cask mactex --zap`. See `brew help rm` for more.

**Reply by SMillerDev** [reply]
That will remove the package, which is not what was asked

**Comment by kerim** [comment]
Why hasn't this been resolved yet? 

Looks like it has been around for 6 years? https://apple.stackexchange.com/questions/359456/brew-cleanup-doesnt-scrub-the-caskroom

**Reply by p-linnane** [reply]
> Why hasn't this been resolved yet?

Because no one has opened a PR to propose any kind of change here. Be the change you want to see.

**Reply by rtsui-harmonicinc** [reply]
> So apparently in this issue [github.com/Homebrew/homebrew-cask/issues/11099](https://github.com/Homebrew/homebrew-cask/issues/11099) they argued that the reason is that you may want to downgrade/fix a package by reinstalling a previous version. Still I think that's a stretch and there's no reason you shouldn't be able to download an old version directly when it's needed. Keeping the files around feels like a huge waste of space. – 
[xji](https://apple.stackexchange.com/users/49307/xji)
Commented Aug 3, 2021 at 0:01

**Reply by SMillerDev** [reply]
> there's no reason you shouldn't be able to download an old version directly when it's needed.

A lot of distributors will always replace the same file they upload. So only the latest file can be downloaded from the URL.
